@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -341,11 +341,18 @@ export function placePointLabel(
 ): PlacementResult {
     const layoutStyle = labelState.element.layoutStyle!;
 
-    // For centered point labels do only current anchor testing.
+    // For the new labels with rejected icons we don't need to go further.
+    // Make them invisible.
+    const newLabel = !labelState.visible;
+    if (isRejected && newLabel) {
+        return PlacementResult.Invisible;
+    }
+    // For centered point labels and labels with icon rejected, do only current anchor testing.
     // TODO: Should be removed after placements are provided via style - see HARP-6487.
     if (
-        layoutStyle.verticalAlignment === VerticalAlignment.Center &&
-        layoutStyle.horizontalAlignment === HorizontalAlignment.Center
+        isRejected ||
+        (layoutStyle.verticalAlignment === VerticalAlignment.Center &&
+            layoutStyle.horizontalAlignment === HorizontalAlignment.Center)
```

Comment:
Since you're handling rejected labels here (both persistent and new ones), you don't need to pass the rejected flag to placePointLabelChoosingAnchor.

Reply:
Probably you are thinking about caller function (`placePointLabel`). I agree that code there already excludes such scenario (rejected icon), but because I wanted it to be kind of independent function that may be used also for different scenarios I left rejected handling here. It is not that easy to understand that flow for further developers, so I am sure it is worth to leave this function flexible and compatible to its counterpart  `placePointLabelAtCurrentAnchor` as the name suggest.

Reply:
So far `placePointLabelAtCurrentAnchor` is just an auxiliary function that is only used by placePointLabelChoosingAnchor. So better remove what's not needed, later it can be added if necessary.

Reply:
Done.